standalone-metastore/metastore-common/src/main/java/org/apache/hadoop/hive/metastore/HiveMetaStoreClient.java

```diff
@@ -2188,12 +2186,12 @@ public List<Partition> listPartitions(String catName, String db_name, String tbl
     if (db_name == null || tbl_name == null) {
       throw new MetaException("Database name/Table name should not be null");
     }
-    boolean skipColumnSchemaForPartition = MetastoreConf.getBoolVar(conf, ConfVars.METASTORE_CLIENT_FIELD_SCHEMA_FOR_PARTITIONS);
```

Comment:
Shouldn't we do similar changes for add_partitions() and drop_partitions() APIs?
I think offloading these APIs reduces the load on clients.

Reply:
For `add_partitions`, it would be a bit confusing as the partitions returned from HMS are different from the partitions to be added, e.g, miss some partition parameters.
For `drop_partitions`, it's debatable, as the PR mainly focuses on the Get operations, what do you think of?

Reply:
Another point is that we will send the partition details along with the notification event:
https://github.com/apache/hive/blob/master/standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/HMSHandler.java#L5287

I'm not sure if it will introduce some surprises to the notification listener.

Reply:
Agreed!! We can do some future work on this if it is required.